x-pack/platform/plugins/shared/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -384,9 +384,30 @@ function buildClaimPartitions(opts: BuildClaimPartitionsOpts): ClaimPartitions {
       continue;
     }
 
-    const capacity = getCapacity(definition.type) / definition.cost;
-    if (capacity !== 0) {
-      result.limitedTypes.set(definition.type, capacity);
+    // task type has maxConcurrency defined
+
+    const isSharingConcurrency = sharedConcurrencyTaskTypes(type);
+    if (isSharingConcurrency) {
+      let minCapacity = null;
+      for (const sharedType of isSharingConcurrency) {
+        const def = definitions.get(sharedType);
+        if (def) {
+          const capacity = getCapacity(def.type) / def.cost;
+          if (minCapacity == null) {
+            minCapacity = capacity;
+          } else if (capacity < minCapacity) {
+            minCapacity = capacity;
+          }
+        }
+      }
+      if (minCapacity) {
+        result.limitedTypes.set(isSharingConcurrency.join(','), minCapacity);
```

Comment:
As you noted, this suspect this may be a little confusing - not to me now, since I was already familiar with what's going on.  I think a comment in the `ClaimPartitions` definition would be appropriate.

Can't think of an alternative approach ATM, and this seems good enough.  However, there's an implication that task types won't have an embedded comma in their string.  I think we should add a gate for this in task registration ...